src/libsync/theme.cpp

```diff
@@ -466,64 +466,61 @@ bool Theme::aboutShowCopyright() const
     return true;
 }
 
-QIcon Theme::syncStateIcon(SyncResult::Status status, bool sysTray, bool sysTrayMenuVisible) const
-{
-    return syncStateIcon(SyncResult { status }, sysTray, sysTrayMenuVisible);
-}
-QIcon Theme::syncStateIcon(const SyncResult &result, bool sysTray, bool sysTrayMenuVisible) const
-{
-    QString statusIcon;
+namespace {
+    QString syncStateIconName(SyncResult result, bool sysTray, bool sysTrayMenuVisible)
+    {
+        QString statusIcon;
 
-    switch (result.status()) {
-    case SyncResult::NotYetStarted:
-        Q_FALLTHROUGH();
-    case SyncResult::SyncRunning:
-        statusIcon = QStringLiteral("state-sync");
-        break;
-    case SyncResult::SyncAbortRequested:
-        Q_FALLTHROUGH();
-    case SyncResult::Paused:
-        statusIcon = QStringLiteral("state-pause");
-        break;
-    case SyncResult::SyncPrepare:
-        Q_FALLTHROUGH();
-    case SyncResult::Success:
-        if (result.hasUnresolvedConflicts()) {
-            return syncStateIcon(SyncResult { SyncResult::Problem }, sysTray, sysTrayMenuVisible);
+        switch (result.status()) {
+        case SyncResult::NotYetStarted:
+            Q_FALLTHROUGH();
+        case SyncResult::SyncRunning:
+            statusIcon = QStringLiteral("state-sync");
+            break;
+        case SyncResult::SyncAbortRequested:
+            Q_FALLTHROUGH();
+        case SyncResult::Paused:
+            statusIcon = QStringLiteral("state-pause");
+            break;
+        case SyncResult::SyncPrepare:
```

Comment:
the fix up is wrong, you need to move offline else, SyncPrepare will now get the offline icon